pkg/utils/createorupdate.go

```diff
@@ -0,0 +1,162 @@
+/*
+© 2020 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	extendedclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+)
+
+func CreateOrUpdateClusterRole(clientSet *clientset.Clientset, clusterRole *rbacv1.ClusterRole) (bool, error) {
+	_, err := clientSet.RbacV1().ClusterRoles().Create(clusterRole)
+	if err == nil {
+		return true, nil
+	} else if errors.IsAlreadyExists(err) {
+		retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			existingClusterRole, err := clientSet.RbacV1().ClusterRoles().Get(clusterRole.Name, v1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to retrieve pre-existing cluster role %s : %v", clusterRole.Name, err)
+			}
+			clusterRole.ResourceVersion = existingClusterRole.ResourceVersion
+			// Potentially retried
+			_, err = clientSet.RbacV1().ClusterRoles().Update(clusterRole)
+			if err != nil {
+				return fmt.Errorf("failed to update pre-existing cluster role %s : %v", clusterRole.Name, err)
+			}
+			return nil
+		})
+		return false, retryErr
+	}
```

Comment:
I think we should use ` controllerutil.CreateOrUpdate` as is done [here](https://github.com/submariner-io/submariner-operator/blob/master/pkg/controller/submariner/submariner_controller.go#L178) but perhaps that should be in another PR.

Reply:
I tried to do that, but `controllerutil` works in the context of a controller (with a plan client and context), it doesn’t work with a clientSet. I imagine it’s possible to get a client and context from a clientSet, but I haven’t figured out how (yet).

Reply:
right - forgot about that. We can look into that later. It's a little harder to do it generically with client sets.